File: ts_bun/hey.ts
import { greet } from './stuff/hello.ts'

if (process.argv.length <= 2 ) {
    console.log("Please supply names !");
    process.exit(1);
}

for(var name of process.argv.slice(2)) {
    var phrase = greet(name);
    console.log(phrase);
}

File: mono/hey.cs
using System;

// So long as all the classes we use are in the same namespace
//  they will all be accessible within its scope
namespace Hey {
    public class hey
    {
        public static void Main(string[] args)
        {
            if (args.Length == 0) {
                Console.WriteLine("Please supply arguments");
                System.Environment.Exit(1);
            }

            foreach(string name in args) {
                Console.WriteLine (Hello.greet(name));
            }
        }
    }
}
